packages/server-core/src/networking/instance/instance.resolvers.ts

```
@@ -49,6 +56,28 @@ export const instanceDataResolver = resolve<InstanceType, HookContext>({
   id: async () => {
     return uuidv4() as InstanceID
   },
+  projectId: async (value, instance, context) => {
+    try {
+      // Populate projectId from locationId
+      if (instance.locationId) {
+        const locationData = await context.app.service(locationPath).get(instance.locationId)
+        if (locationData) {
```

If there is no locationData then it should throw an error.